finetune/target_models/sequence_labeling.py

```diff
@@ -24,24 +22,47 @@ class SequencePipeline(BasePipeline):
     def __init__(self, config, multi_label):
         super(SequencePipeline, self).__init__(config)
         self.multi_label = multi_label
+        self.empty_counts = {"empty": 0, "labeled": 0}
+
+    def _update_empty_ratio(self, empty):
+        if empty:
+            self.empty_counts["empty"] += 1
+        else:
+            self.empty_counts["labeled"] += 1
+        return self.empty_counts
+
+    @property
+    def empty_ratio(self):
+        # Smoothed to prevent zero division
+        return self.empty_counts["empty"] / (self.empty_counts["labeled"] + 1)
 
     def text_to_tokens_mask(self, X, Y=None, context=None):
-        pad_token = [self.config.pad_token] if self.multi_label else self.config.pad_token
+        pad_token = (
+            [self.config.pad_token] if self.multi_label else self.config.pad_token
+        )
         out_gen = self._text_to_ids(X, pad_token=pad_token)
+
         for out in out_gen:
             feats = {"tokens": out.token_ids}
             if context is not None:
                 tokenized_context = tokenize_context(context, out, self.config)
-                feats['context'] = tokenized_context
+                feats["context"] = tokenized_context
             if Y is None:
                 yield feats
             if Y is not None:
                 min_starts = min(out.token_starts)
                 max_ends = max(out.token_ends)
                 filtered_labels = [
-                    lab for lab in Y if lab["end"] >= min_starts and lab["start"] <= max_ends
+                    lab
+                    for lab in Y
+                    if lab["end"] >= min_starts and lab["start"] <= max_ends
                 ]
-                if self.config.filter_empty_examples and len(filtered_labels) == 0:
+                empty = len(filtered_labels) == 0
+                self._update_empty_ratio(empty)
```

Comment:
Only update if the model is trained on a sample. Otherwise it is logging the true data distribution and not the trained data distribution. Therefore you never train any empty samples.